pkg/arvo/sys/vane/ames.hoon

```diff
@@ -2017,7 +2038,22 @@
       ::
       =?  u.cached-state  ?=(%18 -.u.cached-state)
         19+(state-18-to-19:load:adult-core +.u.cached-state)
-      ?>  ?=(%19 -.u.cached-state)
+      =^  moz  u.cached-state
+        ?.  ?=(%19 -.u.cached-state)  [~ u.cached-state]
```

Comment:
shouldn't this preserve `moz` from the earlier upgrade, rather than shadowing those moves with `~`?  This applies to the branch with no unix-duct too.

Reply:
Let's see... I think that only happens when we are not doing a state migration (going from %n-1 to %n—%19 to %20—but rather when the larval stage is called with the current state %n = %20). If we are coming from an older version in the cached-state we are going to go through all state migrations in sequence so we will never go into the `?.` branches.

Reading the `+load` arm, it looks like that's never going to happen because we never regress to being a larva from the last adult ames-state... unless we are a larva which will mean we are booting from the first time, and then either we will trigger those timers (`public-keys`, `recork`...) `+on-init`, or in the case of the `%saxo` in `+on-born`.

Reply:
But the case of the no unix-duct, yeah you're right, we need to preserve moz; thanks!